foreshadow/concrete/internals/cleaners/customizable_base.py

```diff
@@ -0,0 +1,57 @@
+"""Customizable BaseCleaner for cleaner transformers that user can extend."""
+
+from .base import BaseCleaner, CleanerReturn
+
+
+class CustomizableBaseCleaner(BaseCleaner):
+    """Base class for any user customizable cleaner transformer.
+
+    This experimental class is to hide the match_len field from the user as I
+    feel this is rather internal and tedious to explain if user ask about it.
+
+    With this class, the user provided transformation function only needs to
+    consider the transformation logic without wondering what that match length
+    means.
+
+    I want to refactor this metric calculation but don't want to do anything
+    hasty at this moment just for the demo.
+    """
+
+    def transform_row(self, row_of_feature, return_tuple=True):
```

Comment:
As discussed, it might make more sense to the default for return_tuple to False.

Reply:
On second thought, we may need the return_tuple=True for internal computing purpose. The user won't touch this part when supplying the cutomized class so it should be fine.